### PR TITLE
feat: set up kubernetes api server for each machine

### PR DESCRIPTION
**Pull request by Unix4ever**
Make it bind on the siderolink address.
Pull in secrets generation code from the Talos repo.

Set up a single embedded etcd for the whole emulator. Each API server connects to the same etcd but is using a different prefix: one prefix per cluster.

Implement controllers that clean up etcd state after the cluster is destroyed.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -31,53 +42,171 @@ require (
...
 	google.golang.org/grpc v1.65.0
 	google.golang.org/protobuf v1.34.2
+	k8s.io/apimachinery v0.30.2
+	k8s.io/apiserver v0.30.2
+	k8s.io/client-go v0.30.2
+	k8s.io/kubernetes v1.30.2
 )
 
 require (
+	cloud.google.com/go/compute/metadata v0.3.0 // indirect
```
Comment (Unix4ever) [inline go.mod:71]: Unfortunately Kubernetes pulled lots of shit

**Review by smira — APPROVED** [review]
:cool:

File: go.mod [hunk 2]
```diff
@@ -2,16 +2,27 @@ module github.com/siderolabs/talemu
 
 go 1.22.5
 
-// forked go-yaml that introduces RawYAML interface, which can be used to populate YAML fields using bytes
-// which are then encoded as a valid YAML blocks with proper indentation
-replace gopkg.in/yaml.v3 => github.com/unix4ever/yaml v0.0.0-20220527175918-f17b0f05cf2c
+replace (
+	// forked go-yaml that introduces RawYAML interface, which can be used to populate YAML fields using bytes
+	// which are then encoded as a valid YAML blocks with proper indentation
+	gopkg.in/yaml.v3 => github.com/unix4ever/yaml v0.0.0-20220527175918-f17b0f05cf2c
```
Comment (frezbo) [inline go.mod:8]: can't we use the project that implements a basic apiserver, rather than importing all k8s? i can't recollect the project name

Comment (Unix4ever) [reply go.mod:8]: it was `sample-apiserver`, I've tried that one. It's missing lots of things required for the core API server which are not documented anywhere.
So I thought that it's easier to just run a real API server, which sets up everything properly.

Comment (frezbo) [reply go.mod:8]: not that one, we can reconsider once I find the project again :sweat_smile:

**Pull request merged**